### PR TITLE
chore: Generate file descriptor set constant directly

### PR DESCRIPTION
**Pull request by tottoto**
Changes to generate file descriptor set constants directly instead of exporting them as files and reincluding them.

### CODE REVIEW

File: codegen/src/main.rs [hunk 1]
```diff
@@ -117,8 +118,19 @@ fn codegen(
...
+    let fds_raw = buf.as_ref();
+    let tokens = quote! {
+        /// Byte encoded FILE_DESCRIPTOR_SET.
+        pub const FILE_DESCRIPTOR_SET: &[u8] = &[#(#fds_raw),*];
+    };
+
+    let content = format!("{GENERATED_COMMENT}\n{tokens}\n",);
+
+    let mut writer = BufWriter::new(File::create(path).unwrap());
+    writer.write_all(content.as_bytes()).unwrap();
```
Comment (djc) [inline codegen/src/main.rs:135]: Maybe do some formatting, either with prettyplease or `rustfmt`?

File: codegen/Cargo.toml [hunk 2]
```diff
@@ -10,3 +10,4 @@ version = "0.1.0"
 tempfile = "3.8.0"
 protox = "0.7"
 tonic-build = {path = "../tonic-build", default-features = false, features = ["prost", "cleanup-markdown"]}
+quote = "1"
```
Comment (djc) [inline codegen/Cargo.toml:13]: Nit: keep these in alphabetic order?

**Review by djc — APPROVED** [review]
Nice!

File: codegen/src/main.rs (hunk 1 above)
Comment (tottoto) [reply codegen/src/main.rs:136]: I've considered and tried formatting, but in this case, it does not seem helpful for up as they are mere slices of bytes resulting in a lot of newlines by each elements.

File: codegen/Cargo.toml (hunk 2 above)
Comment (tottoto) [reply codegen/Cargo.toml:11]: Applied the suggestion.

File: codegen/src/main.rs (hunk 1 above)
Comment (djc) [reply codegen/src/main.rs:136]: Okay. IMO reviewing future changes to these files will probably still be easier if they're formatted more like "normal" Rust files.

Comment (tottoto) [reply codegen/src/main.rs:137]: Sounds reasonable to me. Changed to format them.

**Review by djc — APPROVED** [review]

**Comment by GitHub23114** [comment]
> Changes to generate file descriptor set constants directly instead of exporting them as files and reincluding them.

Ghost

**Pull request merged**